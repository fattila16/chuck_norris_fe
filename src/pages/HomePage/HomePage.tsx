import React from 'react';
import RandomFacts from '../../components/RandomFacts/RandomFacts';
import useRandom from '../../hooks/useRandom';
import { useRouter } from 'next/router';

const HomePage: React.FC = () => {
    const router = useRouter();
    const { data: randomFact, error, refresh } = useRandom(router?.query?.category);
    return (
        <>
            <RandomFacts randomFact={randomFact} error={error} refresh={refresh} />
        </>
    );
};
export default HomePage;
