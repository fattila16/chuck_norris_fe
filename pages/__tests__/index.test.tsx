import React from 'react';
import { shallow } from 'enzyme';
import HomePage from '../index';

describe('[HomePage]', () => {
    it('Should render HomePage component without error', () => {
        const component = shallow(<HomePage />);
        expect(component.getElements()).toMatchSnapshot();
    });
});
