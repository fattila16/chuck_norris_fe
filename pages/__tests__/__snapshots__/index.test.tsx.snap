// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[HomePage] Should render HomePage component without error 1`] = `
Array [
  <React.Fragment>
    Hello Chuck Norris Facts.
  </React.Fragment>,
]
`;
